dcs-packaging-tool/dcs-packaging-tool-gui/src/main/java/org/dataconservancy/packaging/gui/view/impl/PackageMetadataViewImpl.java

```diff
@@ -0,0 +1,542 @@
+                textField.setId(packageMetadata.getName());
+                allFields.add(textField);
+
+                if (packageMetadata.getValidationType().equals(PackageMetadata.ValidationType.PHONE)) {
+                    HBox horizontalBox = createHBoxForType(textField, PackageMetadata.ValidationType.PHONE);
+                    fieldContainer.getChildren().add(horizontalBox);
+                } else if (packageMetadata.getValidationType().equals(PackageMetadata.ValidationType.EMAIL)) {
+                    HBox horizontalBox = createHBoxForType(textField, PackageMetadata.ValidationType.EMAIL);
+                    fieldContainer.getChildren().add(horizontalBox);
+                } else {
+                    fieldContainer.getChildren().add(textField);
+                }
+            }
+        }
+        return fieldContainer;
+    }
+
+    /**
+     * Helper method to createFields helper method to create the HBox for fields that need validation.
+     *
+     * @param textField
+     * @param validationType
+     * @return container hbox with the validation
+     */
+    private HBox createHBoxForType(TextField textField, PackageMetadata.ValidationType validationType) {
+        HBox horizontalBox = new HBox();
+        Label inputVerificationLabel = new Label();
+        inputVerificationLabel.setPadding(new Insets(3, 0, 0, 3));
+        if (validationType.equals(PackageMetadata.ValidationType.URL)) {
+            textField.textProperty().addListener(getNewChangeListenerForUrl(textField.getId(), inputVerificationLabel));
+        } else if (validationType.equals(PackageMetadata.ValidationType.PHONE)) {
+            textField.textProperty().addListener(getNewChangeListenerForPhoneNumber(textField.getId(), inputVerificationLabel));
+        } else if (validationType.equals(PackageMetadata.ValidationType.EMAIL)) {
+            textField.textProperty().addListener(getNewChangeListenerForEmail(textField.getId(), inputVerificationLabel));
+        }
+        horizontalBox.getChildren().add(textField);
+        horizontalBox.getChildren().add(inputVerificationLabel);
+        return horizontalBox;
+    }
+
+    /**
+     * Helper method that creates labels with tooltips for a given text.
+     *
+     * @param text
+     * @return container vbox
+     */
+    private VBox createSectionLabel(String text) {
+        VBox labelVBox = new VBox();
+        Label label = new Label(text);
+        label.getStyleClass().add(FORM_FIELDS_DIVISION_CLASS);
+        labelVBox.getChildren().add(label);
+        Separator separator = new Separator(Orientation.HORIZONTAL);
+        labelVBox.getChildren().add(separator);
+        return labelVBox;
+    }
+
+    /**
+     * Listener to validate phone number.
+     *
+     * @param errorMessageLabel
+     * @return ChangeListener
+     */
+    private ChangeListener<String> getNewChangeListenerForPhoneNumber(final String packageMetadataName, final Label errorMessageLabel) {
```

Comment:
These should really use our PropertyValidationListener util class. If necessary that class can get a new constructor that fits the needs of this view.